react/FormSubmit.tsx

```diff
@@ -16,12 +16,21 @@ function FormSubmit(props: Props) {
     submitButtonLabel = 'store/newsletter-submit.submitButtonLabel.default',
   } = props
 
-  const { submission } = useNewsletterState()
+  const { submission, confirmation } = useNewsletterState()
   const handles = useCssHandles(CSS_HANDLES)
 
+  // Makes sure the submit button cannot be pressed if there is a
+  // 'newsletter-checkbox-confirmation' in the newsletter form and
+  // it's not checked.
+  const shouldBeDisabled = confirmation === false
+
   return (
     <div className={handles.formSubmitContainer}>
-      <Button type="submit" isLoading={submission.loading}>
+      <Button
+        disabled={shouldBeDisabled}
```

Comment:
I think it should trigger the error message of the checkbox instead of disabling the button

Reply:
It looks like the `Checkbox` component doesn't have an `error` prop, as the `Input` does.
Do you think it makes sense to handle that in the `FormComfirmationCheckbox` component itself then?

Also, I'm not sure how I feel about disabling the button or not... how do you feel about it?

Reply:
That's the right UX pattern, otherwise the user dont' know what's missing to fill.